languages/js/src/types.ts

```diff
@@ -612,10 +609,7 @@ export function isIterable(x: unknown): x is Iterable<unknown> {
  * @internal
  */
 export function isAsyncIterable(x: unknown): x is AsyncIterable<unknown> {
-  return (
-    Symbol.asyncIterator in Object(x) &&
-    typeof Object(x)[Symbol.asyncIterator] === 'function'
-  );
+  return isObj(x) && Symbol.asyncIterator in x;
```

Comment:
do we need to check `typeof x[Symbol.asyncIterator] === 'function'`?

Reply:
I'm not sure... the only reason it wouldn't be a function is if someone fiddled around with it manually. If they fiddled with it, then the runtime error would be `TypeError: x is not iterable` if we tried to use it in a `for...of` or `for await...of` loop or `yield` from it. I'm kind of inclined to let it blow up since there's only so much validation we can be expected to do. What do you think?

Reply:
Updated these type guards a bit in 959afd09, but the update didn't address your question.